config/zookeeper_persister_test.go

```diff
@@ -179,14 +175,60 @@ func TestHistoricalConfigs(t *testing.T) {
 		t.Fatalf("Historical configs are not correct: %+v", cfgs)
 	}
 
-	newConfig, err := Unmarshal(cfgs[0])
-	helpers.CheckError(t, err)
+	newConfig := unmarshallOrPanic(cfgs[0])
 
 	if newConfig.Namespaces["test"] == nil {
 		t.Errorf("Config is not valid: %+v", newConfig)
 	}
 }
 
+func TestReadingStaleVersions(t *testing.T) {
+	conn, _, err := zk.Connect(servers, sessionTimeout)
+	helpers.PanicError(err)
+
+	defer conn.Close()
+
+	p, err := NewZkConfigPersister("/conflicting", servers)
+	helpers.CheckError(t, err)
+
+	defer p.(*ZkConfigPersister).Close()
+
+	origCfg := NewDefaultServiceConfig()
+	origCfg.Namespaces["test"] = NewDefaultNamespaceConfig("test")
+	origCfg.Version = 200
+
+	persistOrPanic(p, origCfg)
+
+	// Two changes
+	cfg1 := cloneConfig(origCfg)
+	cfg1.Namespaces["test_new"] = NewDefaultNamespaceConfig("test_new")
+	cfg1.Version = origCfg.Version + 1
+
+	cfg2 := cloneConfig(origCfg)
+	cfg2.Namespaces["test_newer"] = NewDefaultNamespaceConfig("test_newer")
+	cfg2.Version = origCfg.Version + 2
+
+	// Persist in quick succession
+	persistOrPanic(p, cfg1)
+	persistOrPanic(p, cfg2)
+
+	// Wait for callbacks to re-read persisted configs
+	<- p.ConfigChangedWatcher()
+
+	r, err := p.ReadPersistedConfig()
+	helpers.CheckError(t, err)
+
+	latest := unmarshallOrPanic(r)
+
+	if latest.Version != 202 {
+		t.Logf("Latest config should have version 202, but was %v. Persister latest hash == %v; historical == %+v",
+			latest.Version, p.(*ZkConfigPersister).config, reflect.ValueOf(p.(*ZkConfigPersister).configs).MapKeys())
+
+		// TODO(manik) this is a known problem that needs addressing.
```

Comment:
Ok cool. So we're just adding this test for now, and you're going to follow up in a subsequent PR, yeah?

Reply:
Yep